common/lib/xmodule/xmodule/conditional_module.py

```diff
@@ -148,16 +190,29 @@ class ConditionalModule(ConditionalFields, XModule, StudioEditableModule):
         'voted': 'voted'  # poll_question attr
     }
 
-    @lazy
-    def required_modules(self):
-        return [self.system.get_module(descriptor) for
-                descriptor in self.descriptor.get_required_module_descriptors()]
+    def __init__(self, *args, **kwargs):
+        """
+        Create an instance of the Conditional XBlock.
+        """
+        super(ConditionalBlock, self).__init__(*args, **kwargs)
+        # Convert sources xml_attribute to a ReferenceList field type so Location/Locator
+        # substitution can be done.
+        if not self.sources_list:
+            if 'sources' in self.xml_attributes and isinstance(self.xml_attributes['sources'], six.string_types):
+                self.sources_list = [
+                    # TODO: it is not clear why we are replacing the run here (which actually is a no-op
+                    # for old-style course locators. However, this is the implementation of
+                    # CourseLocator.make_usage_key_from_deprecated_string, which was previously
+                    # being called in this location.
+                    BlockUsageLocator.from_string(item).replace(run=self.location.course_key.run)
```

Comment:
I suspect this is an artifact leftover from the initial introduction of opaque keys and not really relevant, but I also think it's fine to leave it as is with this comment. We're not looking to improve on it or explore new and interesting edge case bugs.